core/modules/wbase/SendChannelShared.cc

```diff
@@ -180,19 +178,21 @@ bool SendChannelShared::_transmit(bool erred, bool largeResult, qmeta::CzarId cz
             bool metaSet = _sendChannel->setMetadata(_metadataBuf.data(), _metadataBuf.size());
             if (!metaSet) {
                 LOGS(_log, LOG_LVL_ERROR, "Failed to setMeta " << idStr);
-                kill(streamLock);
+                kill(streamLock, "metadata");
                 return false;
             }
         }
 
         // Put the data for the transmit in a StreamBuffer and send it.
         auto streamBuf = xrdsvc::StreamBuffer::createWithMove(thisTransmit->dataMsg);
         {
+            // Limit the number of concurrent transmits.
+            wcontrol::TransmitLock transmitLock(*_transmitMgr, scanInteractive, largeResult, czarId);
```

Comment:
Where the limit is set?
It would be good to leave some info on this subject here.